### PR TITLE
[vcpkg] Environment Variable Passthrough

### PR DESCRIPTION
**Pull request by cbezault**
Currently on Windows the only environment variables available to the portfile are those found on a whitelist hard-coded into vcpkg. This change allows us to set the variables passed through as a Cmake list in the triplet.

A change should be made to vcpkg to sanitize the linux environment.

### CODE REVIEW

File: toolsrc/src/vcpkg/build.cpp [hunk 1]
```diff
@@ -285,23 +287,43 @@ namespace vcpkg::Build
...
+                                                  const Triplet& triplet)
     {
-        const Triplet& triplet = config.triplet;
-
-        const std::vector<std::string> dep_strings =
-            Util::fmap_flatten(config.feature_list, [&](std::string const& feature) -> std::vector<std::string> {
-                if (feature == "core")
+        return Util::fmap_flatten(feature_list,
+            [&](std::string const& feature) -> std::vector<Features> {
+            if (feature == "core")
```
Comment (vicroms) [inline toolsrc/src/vcpkg/build.cpp]: Small nitpick: Indentation

File: toolsrc/src/vcpkg/build.cpp [hunk 2]
```diff
@@ -353,55 +375,32 @@ namespace vcpkg::Build
         return concurrency;
     }
 
-    static ExtendedBuildResult do_build_package(const VcpkgPaths& paths,
-                                                const PreBuildInfo& pre_build_info,
-                                                const PackageSpec& spec,
-                                                const std::string& abi_tag,
-                                                const BuildPackageConfig& config)
+    static std::vector<System::CMakeVariable> get_cmake_vars(const VcpkgPaths& paths,
+                                                             const PreBuildInfo& pre_build_info,
```
Comment (vicroms) [inline toolsrc/src/vcpkg/build.cpp]: `pre_build_info` is not used in the function

File: toolsrc/src/vcpkg/build.cpp [hunk 3]
```diff
@@ -32,32 +32,21 @@ namespace vcpkg::Build::Command
     using Dependencies::InstallPlanAction;
     using Dependencies::InstallPlanType;
 
-    static constexpr StringLiteral OPTION_CHECKS_ONLY = "--checks-only";
-
     void perform_and_exit_ex(const FullPackageSpec& full_spec,
                              const SourceControlFileLocation& scfl,
                              const ParsedArguments& options,
```
Comment (vicroms) [inline toolsrc/src/vcpkg/build.cpp]: `vcpkg::Util::unused(options)`

**Comment by vicroms** [comment]
@cbezault 

Do you consider necessary to remove `VCPKG_KEEP_ENV_VARS` implemented in PR #5650 and only allow passing environment variables through a triplet file?

File: toolsrc/src/vcpkg/build.cpp (hunk 1 above)
Comment (cbezault) [reply toolsrc/src/vcpkg/build.cpp]: Yeah, I noticed that in the next PR.

File: toolsrc/src/vcpkg/build.cpp (hunk 2 above)
Comment (cbezault) [reply toolsrc/src/vcpkg/build.cpp]: Ah true, but it will be 😄, I'll get rid of it for this PR.

File: toolsrc/src/vcpkg/build.cpp (hunk 3 above)
Comment (cbezault) [reply toolsrc/src/vcpkg/build.cpp]: Ok.

**Comment by cbezault** [comment]
No I don't think it's necessary, the two can live together.

**Review by Rastaban — APPROVED** [review]
I think I spent more time sorting through all the unrelated cleanup than reviewing the actual passthrough change.  We should get this checked in so it cleans up your other related PRs :)

**Pull request merged**